src/words/environment/WordsEnvironment.java

```diff
@@ -54,31 +55,40 @@ public WordsClass getClass(String className) {
 	 * Creates a new object in the environment and returns it.  Throws an exception if the object could not be created.
 	 */
 	public WordsObject createObject(String objectName, String className, WordsPosition position) throws WordsRuntimeException {
-		if (getObject(objectName) != null) {
+		try {
+			getObject(objectName);
 			throw new WordsObjectAlreadyExistsException(objectName);
-		}
-		
-		WordsClass wordsClass = getClass(className);
+		} catch (WordsObjectNotFoundException e){
 		
-		if (wordsClass == null) {
-			throw new WordsClassNotFoundException(className);
+			WordsClass wordsClass = getClass(className);
+			
+			if (wordsClass == null) {
+				throw new WordsClassNotFoundException(className);
```

Comment:
I think that WordsEnvironment.getClass should probably throw the class not found exception, instead of checking for null here. That's the pattern we established with .getObject